python/plugins/processing/gui/ProcessingToolbox.py

```diff
@@ -211,11 +229,17 @@ def executeAlgorithm(self):
                 dlg.exec_()
                 return
 
+            if self.in_place_mode and len(alg.parameterDefinitions()) <= 2:
+                # hack
```

Comment:
better fix this hack -- it should be: 

- get names of all parameters
- remove input/output names
- if none left, execute_in_place